### PR TITLE
Change the style of date value in Safari to match placeholder "look"

### PR DESCRIPTION
**Pull request by mzbeetnoff**
Resolves:
https://moveinc.atlassian.net/browse/AT-4361

In Safari, date inputs are displayed using a generated placeholder value that is today's date. With the current style system, it wasn't clear to users if the placeholder text was actually an entered value or not. This caused even more confusion when they would go to submit a form and they would receive an invalid date error.

This change explicitly styles the text to look like a placeholder when the input has no value, and to look like an entered value when the input contains some value that is not "".

Safari:

https://user-images.githubusercontent.com/106633502/187562936-95230321-61c4-4af6-8e3d-01a910cb85e7.mov

Chrome:

Screen Shot 2022-08-30 at 4 37 06 PM

### CODE REVIEW

File: packages/input/src/index.js [hunk 1]
```diff
@@ -118,6 +118,9 @@ function Input(
             "&::-webkit-calendar-picker-indicator:hover + svg": {
               color: "ui_300",
             },
+            '&[type="date"]': {
+              color: ref && ref.current?.value ? "ui_900" : "ui_500",
```
Comment (quaardvark) [inline packages/input/src/index.js:122]: Isn't `ref` not guaranteed? We can probably create a local `ref` and merge it with forwarded using `mergeRefs` from `@rent_avail/utils` package to guarantee it

Comment (mzbeetnoff) [reply packages/input/src/index.js]: Hey thanks for the idea, what a cool method `mergeRefs` 👍

Comment (mzbeetnoff) [reply packages/input/src/index.js]: I don't think we can guarantee that a consumer will provide a `ref` property.

**Review by quaardvark — APPROVED** [review]
Can't really comment on the colors, since I think it's a Design System domain, but I can kinda see a potential issue in `[type=date][disabled]` color looking the same as `[type=date]:not([disabled])`

LGTM otherwise

**Review by pedroroque — APPROVED** [review]
LGTM

**Comment by pkrawc** [comment]
Holy shit safari, WHYYYYYY?!

File: packages/input/src/index.js [hunk 2]
```diff
@@ -118,6 +119,9 @@ function Input(
             "&::-webkit-calendar-picker-indicator:hover + svg": {
               color: "ui_300",
             },
+            '&[type="date"]': {
+              color: inputRef.current?.value ? "ui_900" : "ui_500",
+            },
```
Comment (pkrawc) [inline packages/input/src/index.js:124]: Do we need to use a ref here at all? Don't we have the inner value we can rely on?

Comment (pkrawc) [reply packages/input/src/index.js:124]: No I see, we take that + the fact that it's a date input and make a boolean out of them.

**Comment by pkrawc** [comment]
LTGM

**Review by pkrawc — APPROVED** [review]

**Review by pkrawc — APPROVED** [review]

**Comment by pkrawc** [comment]
Not super sure why it wants to bump the popover package but I'll allow it I guess?

**Comment by mzbeetnoff** [comment]
> Not super sure why it wants to bump the popover package but I'll allow it I guess?

I'm not sure either.

**Comment by pkrawc** [comment]
Could be dependabot bumped something and now lerna wants all the packages
that consume popover to update as a result. It's probably fine.

**Pull request merged**